src/lein_monolith/plugin.clj

```diff
@@ -4,76 +4,93 @@
   (:require
     [lein-monolith.config :as config]
     [lein-monolith.dependency :as dep]
+    [leiningen.core.eval :as eval]
     [leiningen.core.main :as lein]
     [leiningen.core.project :as project]))
 
-
 ;; ## Profile Generation
 
+(def profile-config
+  "Configuration for inherited profiles. Structured as a vector of pairs to
+  maintain ordering."
+  [[:monolith/inherited
+    {:types #{}
+     :ks {:inherit :inherit
+          :subproject :monolith/inherit}}]
+
+   [:monolith/inherited-raw
+    {:types #{:raw}
+     :ks {:inherit :inherit-raw
+          :subproject :monolith/inherit-raw}}]
+
+   [:monolith/leaky
+    {:types #{:leaky}
+     :ks {:inherit :inherit-leaky
+          :subproject :monolith/inherit-leaky}}]
+
+   [:monolith/leaky-raw
+    {:types #{:leaky :raw}
+     :ks {:inherit :inherit-leaky-raw
+          :subproject :monolith/inherit-leaky-raw}}]])
```

Comment:
It seems like it would be simpler to flatten this structure and then pass the whole map to whatever functions need it:
```suggestion
    {:monolith :inherit-leaky-raw
     :subproject :monolith/inherit-leaky-raw
     :leaky? true
     :raw? true}]])
```